src/OrchardCore.Modules/OrchardCore.Google/Configuration/GoogleOptionsConfiguration.cs

```diff
@@ -0,0 +1,94 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Google;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Google.Services;
+using OrchardCore.Google.Settings;
+
+namespace OrchardCore.Google.Configuration
+{
+    public class GoogleOptionsConfiguration :
+        IConfigureOptions<AuthenticationOptions>,
+        IConfigureNamedOptions<GoogleOptions>
+    {
+        private readonly IGoogleAuthenticationService _googleAuthenticationService;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly ILogger<GoogleOptionsConfiguration> _logger;
+        private readonly string _tenantPrefix;
+
+
+        public GoogleOptionsConfiguration(
+            IGoogleAuthenticationService googleAuthenticationService,
+            IDataProtectionProvider dataProtectionProvider,
+            ILogger<GoogleOptionsConfiguration> logger,
+            ShellSettings shellSettings)
+        {
+            _googleAuthenticationService = googleAuthenticationService;
+            _dataProtectionProvider = dataProtectionProvider;
+            _logger = logger;
+            _tenantPrefix = "/" + shellSettings.RequestUrlPrefix;
+        }
+
+        public void Configure(AuthenticationOptions options)
+        {
+            var settings = GetGoogleAuthenticationSettingsAsync().GetAwaiter().GetResult();
+            if (settings == null)
+            {
+                return;
+            }
+
+            if (_googleAuthenticationService.ValidateSettings(settings).Any())
+                return;
+
+            options.AddScheme(GoogleDefaults.AuthenticationScheme, builder =>
+            {
+                builder.DisplayName = "Google";
+                builder.HandlerType = typeof(GoogleHandler);
+            });
+        }
+
+        public void Configure(string name, GoogleOptions options)
+        {
+            if (!string.Equals(name, GoogleDefaults.AuthenticationScheme, StringComparison.Ordinal))
+            {
+                return;
+            }
+            var settings = GetGoogleAuthenticationSettingsAsync().GetAwaiter().GetResult();
+            options.ClientId = settings?.ClientID ?? string.Empty;
+            try
+            {
+                options.ClientSecret = _dataProtectionProvider.CreateProtector(GoogleConstants.Features.GoogleAuthentication).Unprotect(settings.ClientSecret);
+            }
+            catch
+            {
+                _logger.LogError("The Consumer Secret could not be decrypted. It may have been encrypted using a different key.");
+            }
+
+            if (settings.CallbackPath.HasValue)
+            {
+                options.CallbackPath = settings.CallbackPath;
+            }
+            //options.SignInScheme = "Identity.External";
```

Comment:
Either leave the comment with some explanation of why there is still a comment, or remove the line.